src/server/pkg/sync/sync.go

```diff
@@ -179,20 +196,56 @@ func (p *Puller) Pull(client *pachclient.APIClient, root string, repo, commit, f
 			})
 		}
 		if emptyFiles {
-			return p.makeFile(path, func(w io.Writer) error { return nil })
+			_, err := p.makeFile(path, false, func(w io.Writer) error { return nil })
+			return err
 		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	limiter := limit.New(concurrency)
```

Comment:
There is only one GetFiles call here, so we don't need to run this in a goroutine. We can get rid of the error group and limiter.

Reply:
Done.